### PR TITLE
handle RTL configuration for iOS and web

### PR DESCRIPTION
**Pull request by camario25**
- added Icon attribute rtlFlip: boolean. This attribute is used to either add a tag to Contents.json for iOS or add class "rlt-flip" for web.
- Tests for both cases.
- for ios in Distribute
- for web in Generate

### CODE REVIEW

File: packages/@icon-magic/generate/src/plugins/svg-generate.ts [hunk 1]
```diff
@@ -68,7 +68,14 @@ export const svgGenerate: GeneratePlugin = {
     const attributes = { id: `${icon.iconName}-${flavor.name}`, 'aria-hidden': true, 'role': 'none'};
     let setCurrentColor = true; // by default, sets the colour of the icon to take the currentColor
 
-    const classNames = params.classNames;
+    let classNames = params.classNames;
+
+    if (classNames && icon.rtlFlip) {
+      classNames.push("rtl-flip");
+    }
+    else if (!classNames && icon.rtlFlip) {
```
Comment (oa495) [inline packages/@icon-magic/generate/src/plugins/svg-generate.ts:76]:
```suggestion
    else if (!classNames) {
```

File: packages/@icon-magic/distribute/src/create-image-set.ts [hunk 2]
```diff
@@ -29,7 +30,7 @@ interface AssetCatalog {
  */
 export async function createImageSet(iconSet: IconSet, outputPath: string) {
   for (const icon of iconSet.hash.values()) {
-    LOGGER.debug(`Creating imageSet for ${icon.iconName}`);
+    LOGGER.debug(`Creating imageSet for ${icon.iconName}, rtl: ${icon.rtlFlip}`);
```
Comment (oa495) [inline packages/@icon-magic/distribute/src/create-image-set.ts:33]:
```suggestion
    LOGGER.debug(`Creating imageSet for ${icon.iconName}, rtl: ${icon.rtlFlip || false}`);
```

don't want it to log undefined

File: packages/@icon-magic/generate/src/plugins/svg-generate.ts [hunk 3]
```diff
@@ -68,7 +68,11 @@ export const svgGenerate: GeneratePlugin = {
     const attributes = { id: `${icon.iconName}-${flavor.name}`, 'aria-hidden': true, 'role': 'none'};
     let setCurrentColor = true; // by default, sets the colour of the icon to take the currentColor
 
-    const classNames = params.classNames;
+    const classNames = params.classNames || [];
+
+    if (icon.rtlFlip) {
```
Comment (oa495) [inline packages/@icon-magic/generate/src/plugins/svg-generate.ts:73]:
```suggestion
    if (icon.rtlFlip && classNames.indexOf("rtl-flip") === -1) {
```

File: packages/@icon-magic/generate/test/svg-generate-test.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ const icon = new Icon({
   sizes: [8, 16],
   resolutions: [1, 2, 3],
   outputPath: `/${FIXTURES}/out`,
+  rtlFlip: true,
```
Comment (oa495) [inline packages/@icon-magic/generate/test/svg-generate-test.ts:35]: you should remove this from here and only add it to one test, the one where you're testing "adds class "rtl-flip" if rtlFlip is true"

Comment (camario25) [reply packages/@icon-magic/generate/test/svg-generate-test.ts:35]: Ok, I'm creating a new test icon `iconFlip`. Thx.

File: packages/@icon-magic/generate/test/fixtures/out/home-2/ads/inverse-small.svg [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="flip" id="ads-inverse-small" aria-hidden="true" role="none" data-supported-dps="24x24">
```
[the remark this answers is not in the document]
Comment (camario25) [reply packages/@icon-magic/generate/test/fixtures/out/home-2/ads/inverse-small.svg:1]: I wouldn't be too concerned with this change.. I learned that the out svg is just the result of the very last test in the test suite.  it used to be testing is class was added and now I added rtl-flip.  But having said that, using the new Icon `iconFlip` and moving the test order should bring back the class flip.. :)

[the remark this answers is not in the document]
Comment (camario25) [reply packages/@icon-magic/generate/test/fixtures/out/home-2/ads/inverse-small.svg:1]: Ok so figured out how this file changes: it's from this temp config:

```
   // Change the config
    const iconrcPath = path.resolve(buildInput, 'iconrc.json');
    const iconrc = JSON.parse(fs.readFileSync(iconrcPath, 'utf8'));
    const tempPlugin = [
      {
        name: 'svg-generate',
        params: {
          addSupportedDps: 'current',
          colorByNameMatching: ['inverse'],
          isFixedDimensions: false, // isFixedDimensions removes the width and height
          classNames: ["flip"]
        }
      }
    ];
    ```
    
    And so you can see that fix width is false so height and width disapear, also there is NO classNames attribute as so there should Not be a a class.  I don't know how there was one before!  but this is the expected behavior..

[the remark this answers is not in the document]
Comment (camario25) [reply packages/@icon-magic/generate/test/fixtures/out/home-2/ads/inverse-small.svg:1]: I confirmed by changing the attributes in this temp config

File: packages/@icon-magic/generate/test/fixtures/out/home-2/ads/active-small.svg [hunk 6]
```diff
@@ -1,4 +1,3 @@
 <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="flip" id="ads-active-small" aria-hidden="true" role="none" data-supported-dps="24x24" fill="currentColor">
   <path d="M12 17.13a5.13 5.13 0 010-10.25V4a8 8 0 108 8h-2.87A5.13 5.13 0 0112 17.13z" style="isolation:isolate" opacity=".75"/>
-  <path d="M23.31 6.34l-4.95-.71-.7-4.94-3.54 3.53.35 2.48-3.89 3.89a2 2 0 002.83 2.83l3.89-3.89 2.48.35z"/>
```
Comment (camario25) [inline packages/@icon-magic/generate/test/fixtures/out/home-2/ads/active-small.svg:3]: Found out these paths come from this test:

```
    // Run generate again, hashing is true BUT the svg has been changed so plugins should run.
    const newSvg = `<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
      <title>24dp</title>
      <path d="M12,17.13A5.13,5.13,0,0,1,12,6.88V4a8,8,0,1,0,8,8H17.13A5.13,5.13,0,0,1,12,17.13Z" style="opacity: 0.75;isolation: isolate"/>
    </svg>`;
    const inputSvgPath = path.resolve(buildInput, `active-small.svg`);
    fs.writeFileSync(inputSvgPath, newSvg);
    ```
    
    So it is expected that the path gets removed!  Again I don't know how this path was not removed previously. I suspect hashing.

**Pull request merged**